### ISSUE TITLE
All lines get indented with nesting

### ISSUE THREAD

**Issue by shalim-work** [chrisbra/vim-xml-ftplugin#14]
I noticed, when editing an XML file, that the automatic indent kept indenting each line beyond what it should be; given this file:

```xml
<and>
<this>
<isnt/>
</this>
</and>
```

If I set &indentexpr to `XmlIndentGet(v:lnum, 1)` (letting the xml.vim file set it, in other words), I get no changes at all when I indent the file (`gg=G`). If I set it to `XmlIndentGet(v:lnum, 0)`, I get:

```xml
<and>
    <this>
        <isnt/>
            </this>
                </and>
```

What I'm looking for is:

```xml
<and>
    <this>
        <isnt/>
    </this>
</and>
```

**Comment by chrisbra** [comment]
cannot reproduce. Have you tried the latest version?

**Comment by shalim-work** [comment]
The indent/xml.vim I'm using has a 'Last Changed' of July 27, 2019 and, other than the fact that one on github here says Feb 04, 2019 and mine has the 'Correctly handle non-tagged data' comment, the rest of the contents are identical.

**Comment by chrisbra** [comment]
I tried with the checkout repository:

    vim --clean --cmd 'set rtp+=$PWD' --cmd ':set packpath+=$PWD' test16.xml

(where test16.xml contains your initial sample data). Then I tried:

`:new foobar.xml`

and entered the content manually. It looked all as expected. Can you verify this happens with a fresh checkout version of this plugin?

**Comment by shalim-work** [comment]
Yeah, that worked. I'll dig into my stuff. Sorry for wasting your time, but I appreciate you helping me narrow it down a bit.

**Comment by chrisbra** [comment]
no need to apologize. Good to figure it out. Thanks and let me know if we should change anything here.

**Comment by shalim-work** [comment]
Just so you know, I couldn't figure out why it used to work and now it doesn't. (It used to work without me being forced to try the `clean` Vim instance.) I've switched to using https://www.vim.org/scripts/script.php?script_id=1211 which, while over fourteen years old, works for me.

I'll update my Vim to the latest patch version in a day or two and see if it was something that changed in the interim, but that's a bit of a shot in the dark.

Thank you for putting up with me! :)

**Comment by chrisbra** [comment]
oh well, would have been good to find out.

**Comment by shalim-work** [comment]
I studied the code for both your and the version I ended up using; they have a lot of the same functions (XmlIndentSynCheck and XmlIndentWithPattern are identical, for example). The only real difference I can spot, unless I’m missing things, is that the one I’m using examines the contents of the file with lines such as:

    elseif match(prev_line, '^\s*<%--') == 0
          \ && match(prev_line, '--%>$') == -1
      let ind = ind + 5
    elseif match(prev_line, '^\s*$') == -1

And yours seems to examine the syntax. In fact, even though XmlIndentSynCheck is defined, the one I’m using never calls it (the one usage is commented out).

Salman
